packages/bvaughn-architecture-demo/src/contexts/PointsContext.tsx

```diff
@@ -52,26 +61,51 @@ export function PointsContextRoot({ children }: PropsWithChildren<{}>) {
   const { recordingId, trackEvent } = useContext(SessionContext);
   const replayClient = useContext(ReplayClientContext);
 
-  // Both high-pri state and transition state should be managed by useLocalStorage,
+  // Both high-pri state and transition state should be managed by useIndexedDB,
   // Else values from other tabs will only be synced to the high-pri state.
-  const [points, setPoints] = useLocalStorage<Point[]>(`${recordingId}::points`, []);
-  const [pointsForAnalysis, setPointsForAnalysis, isPending] = useLocalStorage<Point[]>(
-    `${recordingId}::points::transition`,
-    [],
-    true
-  );
+  const {
+    setValue: setPoints,
+    status: pointsStatus,
+    value: points,
+  } = useIndexedDB<Point[]>({
+    database: POINTS_DATABASE,
+    initialValue: [],
+    recordName: recordingId,
+    storeName: "high-priority",
+  });
+  const {
+    setValue: setPointsForAnalysis,
+    status: pointsForAnalysisStatus,
+    value: pointsForAnalysis,
+  } = useIndexedDB<Point[]>({
+    database: POINTS_DATABASE,
+    initialValue: [],
+    recordName: recordingId,
+    scheduleUpdatesAsTransitions: true,
+    storeName: "transition",
+  });
+
+  const isPending = pointsForAnalysisStatus === "update-pending";
+
+  const prevPointsRef = useRef<Point[]>([]);
 
   const setPointsHelper = useCallback(
-    (action: SetStateAction<Point[]>) => {
-      setPoints(action);
-      setPointsForAnalysis(action);
+    (newPoints: Point[]) => {
+      setPoints(newPoints);
+      setPointsForAnalysis(newPoints);
     },
     [setPoints, setPointsForAnalysis]
   );
 
+  useLayoutEffect(() => {
+    if (pointsStatus === "initialization-complete") {
+      prevPointsRef.current = points!;
+    }
```

Comment:
This seems like it could burn us. If we try to add a point before previous ones load (even though it's probably unlikely, timing wise) we'd read an empty array and override the previous points value.

I think we should just prevent the UI from being enabled until the status was complete, and then...restore the `SetStateAction` logic (no ref based thing)

Reply:
This was also left over from your initial WIP commit :)  

I can review it and see if it can be improved.

Reply:
Just to be clear, the initial WIP commit was very half baked. I didn't finish writing it. I definitely didn't test it. Please feel free to question _anything_ from it.

Reply:
Tried logging some timings here.  In local dev, it's about 100ms from the time the `useIndexedDB` hooks first render to when we have read the data and set state...

and that looks like it's well faster than the time it takes to fetch the initial source file even if it was open on recording load.

Timings are always fickle, but I'm okay with not worrying about code for preventing interactions before the points have been read from IDB.

Reply:
I am pretty uncomfortable with that assumption.

Reply:
Maybe I'm misunderstanding the implications of what you're saying though. Let's chat about it after the sync?